ui/components-react/src/components-react/filter-builder/FilterBuilderState.ts

```diff
@@ -358,7 +365,13 @@ function validateRules(
       const errorMessage = ruleValidator
         ? ruleValidator(item)
         : defaultPropertyFilterBuilderRuleValidator(item);
-      if (errorMessage) ruleIdsAndErrorMessages.set(item.id, errorMessage);
+      if (errorMessage)
+        ruleIdsAndErrorMessages.set(
+          item.id,
+          errorMessage.startsWith("filterBuilder.errorMessages")
```

Comment:
Seems a bit exotic, would it make more sense to wrap/convert `defaultPropertyFilterBuilderRuleValidator` to hook and keep 2 args in `validateRules`?

Reply:
`defaultPropertyFilterBuilderRuleValidator` might be used by FilterBuilder consumers (presentation-components uses it) so it can be simply converted into a hook. Added new `useDefaultPropertyFilterBuilderRuleValidator` and deprecated `defaultPropertyFilterBuilderRuleValidator`. I think it can be removed in 5.0 as it's only a beta. What do you think?

Reply:
Sounds good, we will break some of the existing `@beta` APIs with this PR as well, i.e. `getPropertyFilterBuilderOperatorLabel`, but I assume these are not expected to be used by anyone.

Reply:
Yes, that one I don't expect to be used.